plugin/storage/cassandra/schema/create.go

```diff
@@ -0,0 +1,216 @@
+		}
+	} else {
+		traceTTLMinutes := traceTTL / 60
+		compactionWindowSize = (traceTTLMinutes + 30 - 1) / 30
+		compactionWindowUnit = "m"
+	}
+
+	switch compactionWindowUnit {
+	case "m":
+		compactionWindowUnit = "MINUTES"
+	case "h":
+		compactionWindowUnit = "HOURS"
+	case "d":
+		compactionWindowUnit = "DAYS"
+	default:
+		return 0, "", errors.New("invalid compaction window unit")
+	}
+
+	return compactionWindowSize, compactionWindowUnit, nil
+}
+
+func (mb *MappingBuilder) GetSpanServiceMappings() (spanMapping string, serviceMapping string, err error) {
+	traceTTL, _ := strconv.Atoi(getEnv("TRACE_TTL", "172800"))
+	dependenciesTTL, _ := strconv.Atoi(getEnv("DEPENDENCIES_TTL", "0"))
+	cas_version := getEnv("VERSION", "4")
+	// template := os.Args[1]
+	var template string
+	var cqlOutput string
+	if template == "" {
+		switch cas_version {
+		case "3":
+			template = "./v003.cql.tmpl"
+		case "4":
+			template = "./v004.cql.tmpl"
+		default:
+			template = "./v004.cql.tmpl"
+		}
+		mode := getEnv("MODE", "")
+
+		if mode == "" {
+			fmt.Println("missing MODE parameter")
+			return
+		}
+		var datacentre, replications string
+		// var ReplicationFactor int
+		if mode == "prod" {
+			datacentre = getEnv("DATACENTRE", "")
+			if datacentre == "" {
+				fmt.Println("missing DATACENTRE parameter for prod mode")
+				return
+			}
+			replicationFactor, _ := strconv.Atoi(getEnv("REPLICATION_FACTOR", "2"))
+			replications = fmt.Sprintf("{'class': 'NetworkTopologyStrategy', '%s': %d}", datacentre, replicationFactor)
+		} else if mode == "test" {
+			datacentre = getEnv("DATACENTRE", "")
+			replicationFactor, _ := strconv.Atoi(getEnv("REPLICATION_FACTOR", "1"))
+			replications = fmt.Sprintf("{'class': 'SimpleStrategy', 'replication_factor': '%d'}", replicationFactor)
+		} else {
+			fmt.Printf("invaild mode: %s, expecting 'prod' or 'test'", mode)
+			return
+		}
+		keyspace := getEnv("KEYSPACE", fmt.Sprintf("jaeger_v1_%s", mode))
+		if !isValidKeyspace(keyspace) {
+			fmt.Printf("invaild characters in KEYSPACE=%s parameter , please use letters, digits, and underscores only", keyspace)
+			return
+		}
+		CompactionWindowSize, compactionWindowUnit, _ := getCompactionWindow(traceTTL, getEnv("COMPACTION_WINDOW_SIZE", ""))
+
+		params := MappingBuilder{
+			Mode:                 mode,
+			Datacentre:           datacentre,
+			Keyspace:             keyspace,
+			Replication:          replications,
+			TraceTTL:             traceTTL,
+			DependenciesTTL:      dependenciesTTL,
+			CompactionWindowSize: CompactionWindowSize,
+			CompactionWindowUnit: compactionWindowUnit,
+		}
+		// Render the template
+		cqlOutput, err = params.renderTemplate(template)
+		if err != nil {
+			return "", "", err
+		}
+		cqlOutput, _ = RenderCQLTemplate(params, cqlOutput)
+		// fmt.Println(cqlOutput)
+	}
+	return cqlOutput, "", err
+}
+
+func SchemaInit(c *gocql.Session) {
+	builder := &MappingBuilder{}
+	schema, _, err := builder.GetSpanServiceMappings()
```

Comment:
where does the term "mapping" come from? If you're copying it from Elasticsearch, there "mapping" is actually a term understood by ES. No such thing in Cassandra.

Reply:
should i call it template  or cassandra_template ?

Reply:
SchemaBuilder